lib/cdo/section_helpers.rb

```diff
@@ -1,10 +1,17 @@
 module SectionHelpers
   CHARS = ("A".."Z").to_a - %w(A E I O U)
+
   def self.random_text(len)
     len.times.to_a.collect {CHARS.sample}.join
   end
 
+  # @return [String] A semi-random vowelless string of length six.
   def self.random_code
-    random_text(6)
+    loop do
+      code = random_text(6)
+      # Avoid various naughty substrings.
+      next if %w(CNT DCK DMN FCK PNS PSS SHT TTS).any? {|substring| code.include? substring}
```

Comment:
I'm afraid I don't know much about it; @bcjordan set it up for applab and playlab and I believe the clientside work goes through [this channels api call](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/initApp/project.js#L942-L953)